src/app/snippet/components/snippet-filter/snippet-filter.component.ts

```diff
@@ -0,0 +1,91 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+
+import {Snippet} from '../../models/snippet';
+import {Label} from '../../models/label';
+
+@Component({
+  selector: 'app-snippet-filter',
+  templateUrl: 'snippet-filter.component.html',
+  styleUrls: ['snippet-filter.component.scss']
+})
+export class SnippetFilterComponent implements OnInit {
+
+  buttonBackground = 'linear-gradient(0.44turn, mediumpurple, purple, black)';
+  buttonText = 'Nouveau Snippet';
+
+  @Input()
+  snippets: Snippet[] = [];
+  @Input()
+  labels: Label[] = [];
+  total = 0;
+  favorites = 0;
+  unlabeled = 0;
+  labelsUsedBySnippets = new Map<string, number | undefined>();
+  usedBySnippets: Label[] = [];
+
+  constructor(
+    private router: Router
+  ) {
+  }
+
+  ngOnInit() {
+    this.total = this.snippets.length;
+    this.snippets.forEach(snippet => {
+      if (snippet.favorites) {
+        this.favorites++;
+      }
+      if (snippet.labels === undefined) {
```

Comment:
tu peux le simplifier par `if (!snippet.labels)`, de plus je pense que tu devrais aussi vérifier `this.snippets.length` pour traiter le cas potentiel d'un tableau vide